sdk/core/core-auth/review/core-auth.api.md

```diff
@@ -5,12 +5,14 @@
 ```ts
 
 import { AbortSignalLike } from '@azure/abort-controller';
+import { HttpMethods } from '@azure/core-util';
 
 // @public
 export interface AccessToken {
     expiresOnTimestamp: number;
     refreshAfterTimestamp?: number;
     token: string;
+    tokenType?: string;
```

Comment:
If we're not willing to make this `"Bearer" | "pop"` to get strong typechecking, maybe we could make some new interfaces like `export interface BearerAccessToken extends AccessToken { tokenType: "Bearer" }` so that in Identity we can make sure when we're dealing with bearer or pop tokens we use the right tokenType?

Reply:
I know we've been over this already but I do want to understand a little more about the justification for not just making this a union `"Bearer" | "pop"`. IIUC the reasoning was because we can't use an enum to form the Authorization header value directly?
 
That reasoning holds water for other languages, since they might need extra logic to convert the enum into a string, but not sure it applies for TS, since "Bearer" and "pop" are just string values and can be put straight into the header.

Reply:
My concern is also that msal-node returns a string for `tokenType`

Reply:
The other reasons would be to just leave it as a string is :
- There may be other types added in the future (we really don't know)
- Msal returns a string for `tokenType` - there authentication scheme is of the following -
`export const AuthenticationScheme = {
    BEARER: "Bearer",
    POP: "pop",
    SSH: "ssh-cert",
} as const;`ts

Reply:
I will need to do this everywhere in the identity code if I go with "Bearer" | "pop" type -
` tokenType: (response.tokenType as "Bearer" | "pop")`

It seems like an overkill? I think for the interface we can add as an additive change? @xirzec

Reply:
Actually just changed the return type to cast to `AccessToken`, it worked. Should be enough I guess?

Reply:
@xirzec